File: app/src/main/java/com/example/stiffback/LocationViewModel.java
package com.example.stiffback;

import android.app.Application;
import android.location.Location;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;


public class LocationViewModel extends AndroidViewModel {

    private LocationRepository mRepository;
    
    public LocationViewModel(Application application) {
        super(application);
        mRepository = new LocationRepository(application);
    }

    public final MutableLiveData<LocationCell> getmCompass(){
        return this.mRepository.getmLocationCell();
    }

    public final MutableLiveData<Location> getmLocation(){
        return this.mRepository.getmLocation();
    }

    public void pauseLocationUpdate(){
        mRepository.pauseLocationUpdate();
    }

    public void resumeLocationUpdate(){
        mRepository.resumeLocationUpdate();
    }
}

File: app/src/main/java/com/example/stiffback/LocationViewModel.kt
package com.example.stiffback

import android.app.Application
import android.location.Location

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData


class LocationViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: LocationRepository

    init {
        mRepository = LocationRepository(application)
    }

    fun getmLocationCell(): MutableLiveData<LocationCell> {
        return this.mRepository.getmLocationCell()
    }

    fun getmLocation(): MutableLiveData<Location> {
        return this.mRepository.getmLocation()
    }

    fun pauseLocationUpdate() {
        mRepository.pauseLocationUpdate()
    }

    fun resumeLocationUpdate() {
        mRepository.resumeLocationUpdate()
    }
}